crates/router/src/connector/noon/transformers.rs

```diff
@@ -784,13 +785,18 @@ pub struct NoonPaymentsTransactionResponse {
 }
 
 #[derive(Default, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct NoonRefundResponseResult {
     transaction: NoonPaymentsTransactionResponse,
 }
 
 #[derive(Default, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct RefundResponse {
     result: NoonRefundResponseResult,
+    result_code: u32,
+    class_description: String,
+    message: String,
```

Comment:
Aren't these optional fields? Do we get this in case of refund success scenario also?

Reply:
Yeah. These values are always returned during Refund. In case the refund is successful, empty string or 0 is returned.